### ISSUE TITLE
Fix inconsistent PHP requirements

### ISSUE THREAD

**Issue by jonathanstegall** [MinnPost/object-sync-for-salesforce#13]
We've got places where we use `&$class` to load a class, and maybe it's never necessary. We've also got places where we only use `$class` and maybe that's fine.

Finally, there are places where we use older syntax where a newer syntax might be easier to read.

Ex (in Salesforce Push):

```
$salesforce_field = $fieldmap['salesforce_field'];
$wordpress_field = $fieldmap['wordpress_field'];
$params[$salesforce_field] = $object[$wordpress_field];
```

This could probably be replaced with:

```
$params[$fieldmap['salesforce_field']] = $object[$fieldmap['wordpress_field']];
```

But it only works in newer PHP. Maybe that's fine.

**Comment by jonathanstegall** [comment]
[activate.php](https://github.com/MinnPost/salesforce-rest-api/blob/master/classes/activate.php#L22-L23)

**Comment by jonathanstegall** [comment]
As a whole, this plugin doesn't work in php 5.3, and it's maybe worth finding out exactly what the lowest version it supports is.

It has been tested on 5.3 (fails), 5.6 (works), and 7.0 (works).

**Comment by rclations** [comment]
I ran https://wordpress.org/plugins/php-compatibility-checker/ for php 5.3 to get a sense of what wasn't working, and the list is pretty short:

```FILE: /wp-content/plugins/salesforce-rest-api/classes/salesforce.php
------------------------------------------------------------------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 4 LINES
------------------------------------------------------------------------------------------------------------------------------
 140 | ERROR | Short array syntax (open) is available since 5.4
 140 | ERROR | Short array syntax (close) is available since 5.4
 171 | ERROR | Function array dereferencing is not present in PHP version 5.3 or earlier
 705 | ERROR | Short array syntax (open) is available since 5.4
 707 | ERROR | Short array syntax (close) is available since 5.4
------------------------------------------------------------------------------------------------------------------------------


FILE: /wp-content/plugins/salesforce-rest-api/classes/salesforce_mapping.php
--------------------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
--------------------------------------------------------------------------------------------------------------------------------------
 253 | ERROR | The function array_column is not present in PHP version 5.4 or earlier
 267 | ERROR | The function array_column is not present in PHP version 5.4 or earlier
--------------------------------------------------------------------------------------------------------------------------------------


FILE: /wp-content/plugins/salesforce-rest-api/classes/wordpress.php
-----------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------------------------------------
 456 | ERROR | The function array_column is not present in PHP version 5.4 or earlier
-----------------------------------------------------------------------------------------------------------------------------


FILE: /wp-content/plugins/salesforce-rest-api/templates/admin/user-profile-salesforce-map.php
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 1 | WARNING | No PHP code was found in this file and short open tags are not allowed by this install of PHP. This file may be using short open tags but PHP does not allow them.
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

Not sure that this is a comprehensive list, but do you want me to give it a shot taking care of these issues and see if the plugin will work on 5.3?

**Comment by jonathanstegall** [comment]
I don't feel strongly about this either way. I'd like to establish what the lowest version is so users understand. And if we are getting any benefits by doing things in the newer versions (like if `array_column` is way better than an older alternative) I'd rather keep it in the newer versions.

My original confusion was about the different ways that newer PHP versions handle `&$class` vs `$class` - it looks like the & isn't needed anymore, but things I could find were kind of vague.

**Comment by rclations** [comment]
based on my testing with the PHP Compatibility checker, PHP 5.5 is our minimum requirement. There are some warnings in the vendor folder for all versions, but I don't think those will affect end users

**Comment by jonathanstegall** [comment]
That is fine I think. If we find that people really need it on 5.3 or 5.4, we could perhaps revisit. But let's wait and see if that happens, IMO.

Do you have any thoughts about the & stuff with classes? Are you familiar with that?

**Comment by jonathanstegall** [comment]
This seems to be all resolved with this commit: f1daa0ce21e7289a7c9859b4058d1ce09c459afd
